cogs/roles.py

```diff
@@ -2,227 +2,121 @@
 
-        tmp_emojis = {}
-        emojis = {}
-        degree_program_assignable = self.assignable_roles[0]
+        for role_category, roles in self.assignable_roles.items():
+            prefix = f"assign_{role_category}"
+            self.bot.view_manager.register(prefix, self.on_button_clicked)
 
-        # start with getting all emojis that are used in those roles as a dict
-        for emoji in self.bot.emojis:
-            if emoji.name in degree_program_assignable:
-                tmp_emojis[emoji.name] = emoji
+    def get_stat_roles(self):
+        """ Get all roles that should be part of the stats Command """
 
-        # bring them in desired order
-        for key in degree_program_assignable.keys():
-            emojis[key] = tmp_emojis.get(key)
+        stat_roles = []
+        for category in self.assignable_roles.values():
+            if category["in_stats"]:
+                for role in category["roles"].values():
+                    stat_roles.append(role["name"])
 
-        return emojis
+        return stat_roles
 
-    def get_color_emojis(self):
-        """ Creates a dict for color role emojis """
+    async def on_button_clicked(self, button: disnake.ui.Button, interaction: disnake.MessageInteraction, value=None):
+        """
+        Add or Remove Roles, when Button is clicked. Role gets added, if the user clicking the button doesn't have
+        the role already assigned, and removed, if the role is already assigned
+        """
 
-        emojis = {}
-        color_assignable = self.assignable_roles[1]
+        guild_roles = {str(role.id): role for role in interaction.guild.roles}
+        role = guild_roles.get(value)
 
-        # start with getting all emojis that are used in those roles as a dict
-        for emoji in self.bot.emojis:
-            if emoji.name in color_assignable:
-                emojis[emoji.name] = emoji
-
-        return emojis
-
-    def get_special_emojis(self):
-        """ Creates a dict for special role emojis """
-
-        return self.assignable_roles[2]
-
-    def get_key(self, role):
-        """ Get the key for a given role. This role is used for adding or removing a role from a user. """
-
-        for key, role_name in self.assignable_roles[0].items():
-            if role_name == role.name:
-                return key
-
-    @help(
-        category="info",
-        brief="Gibt die Mitgliederstatistik aus."
-    )
-    @commands.command(name="stats")
-    async def cmd_stats(self, ctx):
-        """ Sends stats in Chat. """
-
-        guild = ctx.guild
+        if role in interaction.author.roles:
+            await interaction.author.remove_roles(role)
+            await utils.send_dm(interaction.author, f"Rolle \"{role.name}\" erfolgreich entfernt")
+        else:
+            await interaction.author.add_roles(role)
+            await utils.send_dm(interaction.author, f"Rolle \"{role.name}\" erfolgreich hinzugefügt")
+
```

Comment:
Mein Testserver ist in der Ansicht relativ weit unten, daher hab ich beim Klick auf den Button gar nicht mitbekommen, dass Boty mir eine neue DM gesendet hat. Bei slash-commands gibt's doch auch die Möglichkeit nur dem momentanen Nutzer, aber direkt im Channel, zu antworten. Das wäre hier evtl. besser.

Reply:
Kommen jetzt als Interaction response im Channel als Ephemeral